meetings/2018-01-17/minutes.md

```diff
@@ -0,0 +1,417 @@
+whether this is something every build tool should define for itself. Stu
+suggests that it should be just "simple extensions" to the LSP protocol. Jon
+asks whether it might be simpler to treat the LSP and BSP as orthogonal
+projects, and Stu and Jorge both agree.
+
+Jorge and Stu debate some of the interactions between LSP and BSP. Stu notes
+that the LSP server does not need a compiler. The discussion goes into some
+technical detail, without reaching a conclusion. Stu would like LSP servers to
+be thinner.
+
+Jon tries with limited success to keep the conversation focussed away from
+technical details.
+
+Eugene Burmako added that he sees SemanticDB as a useful standard that works
+for both complete and partial information. He points out that it doesn't have
+to be persisted, so it can be exchanged in-memory. He notes that it's used to
+index third-party dependencies. It is used to communicate with the presentation
+compiler: the presentation compiler can produce SemanticDB data. Eugene
+suggests that SemanticDB could be a good standard for exchanging information
+between tools.
+
+Jon suggests that the LSP proposal is well-defined, and implementations exist,
+but suggests that the BSP should be designed to be modular to support more
+heterogeneous features. He suggest that different people could be involved with
+working on the two proposals.
+
+Shane Delmore asks about timescales for development, and asks what we're trying
+to support, and mentioned that people may want to work on projects which
+interoperate with other languages. Shane asks whether TASTY works currently
+with Scalac 2.11 or 2.12, and Adriaan says there's currently no schedule for
+including TASTY support in Scalac 2.13. He's open to it, but thinks it's more
+of a Scalac 2.14 target.
+
+Jon claims (in Martin's absence) that Martin didn't think it would be a huge
+amount of work to get a basic implementation of TASTY working for Scalac though
+challenges would exist in deciding the exact representations of Scalac
+structures in a format designed primarily with Dotty in mind, and Adriaan
+thinks that it's a challenge to get TASTY to scale between Java, Scala and
+Dotty.
+
+Jorge suggests some solutions for cross-language support.
+
+## Next Steps
+
+With limited time remaining in the meeting, Jon makes a proposal that two
+working groups should be formed to work on the LSP and the BSP, made up of
+different people with some overlap between the two, and that the working groups
+should come under the remit of the Scala Center.
+
+Jon notes that the work of the Scala Center should be recommended by the
+Advisory Board, and that it would be good to have a proposal from one of the
+sponsors so that an official vote could be taken during a Scala Center Advisory
+Board meeting, proposing that the Scala Center support the initiative.
+
+Jon suggests six to eight people should be in each group.
+
+Stu suggests a "protocols subgroup" within the Scala Center, and suggests that
```

Comment:
To be clear, I suggested a "protocols subgroup" as opposed to a "BSP subgroup", as I want to make sure we've considered all options (including the thin proposal: which I hope to post soon) before deciding that the BSP protocol is the right approach.

Reply:
For the record, I have updated the bsp document to clarify that https://github.com/scalacenter/bsp/blob/master/docs/bsp.md is a "thought experiment for how a potential Build Server Protocol (BSP) would look like". 

I'm very much looking forward to read your thin proposal! I believe this protocol will take a while to mature, we should explore multiple approaches.